packages/account/tests/account.e2e.spec.ts

```diff
@@ -0,0 +1,255 @@
+      signer,
+      biconomyPaymasterApiKey,
+      bundler: new Bundler({
+        bundlerUrl,
+        chainId,
+        entryPointAddress,
+      }),
+    });
+
+    const paymaster = smartWallet.paymaster;
+    expect(paymaster).not.toBeNull();
+    expect(paymaster).not.toBeUndefined();
+  });
+
+  it("Should gaslessly mint an NFT", async () => {
+    const nftAddress: Hex = "0x1758f42Af7026fBbB559Dc60EcE0De3ef81f665e";
+    const {
+      chainId,
+      whale: { signer, publicAddress: recipient },
+      bundlerUrl,
+      entryPointAddress,
+      biconomyPaymasterApiKey,
+      publicClient,
+    } = chainData;
+
+    const smartWallet = await createSmartWalletClient({
+      chainId,
+      signer,
+      biconomyPaymasterApiKey,
+      bundler: new Bundler({
+        bundlerUrl,
+        chainId,
+        entryPointAddress,
+      }),
+    });
+
+    const encodedCall = encodeFunctionData({
+      abi: [
+        {
+          inputs: [{ name: "address", type: "address" }],
+          name: "safeMint",
+          outputs: [{ name: "", type: "uint256" }],
+          stateMutability: "view",
+          type: "function",
+        },
+      ],
+      functionName: "safeMint",
+      args: [recipient],
+    });
+
+    const transaction = {
+      to: nftAddress, // NFT address
+      data: encodedCall,
+      value: 0,
+    };
+    const balance = (await checkBalance(publicClient, recipient, nftAddress)) as bigint;
+    const { wait } = await smartWallet.sendTransaction(transaction, {
+      paymasterServiceData: {
+        mode: PaymasterMode.SPONSORED,
+      },
+    });
+
+    const result = await wait();
+    const newBalance = (await checkBalance(publicClient, recipient, nftAddress)) as bigint;
+
+    expect(newBalance - balance).toBe(1n);
+    expect(result?.receipt?.transactionHash).toBeTruthy();
+  }, 60000);
+
+  it("#getUserOpHash should match entryPoint.getUserOpHash", async () => {
+    const {
+      chainId,
+      whale: { signer },
+      bundlerUrl,
+      entryPointAddress,
+      publicClient,
+      biconomyPaymasterApiKey,
+    } = chainData;
+
+    const smartWallet = await createSmartWalletClient({
+      chainId,
+      signer,
+      biconomyPaymasterApiKey,
+      bundler: new Bundler({
+        bundlerUrl,
+        chainId,
+        entryPointAddress,
+      }),
+    });
+
+    const userOp: UserOperationStruct = {
+      sender: "0x".padEnd(42, "1") as string,
+      nonce: 2,
+      initCode: "0x3333",
+      callData: "0x4444",
+      callGasLimit: 5,
+      verificationGasLimit: 6,
+      preVerificationGas: 7,
+      maxFeePerGas: 8,
+      maxPriorityFeePerGas: 9,
+      paymasterAndData: "0xaaaaaa",
+      signature: "0xbbbb",
+    };
+
+    const epHash = await publicClient.readContract({
+      address: entryPointAddress as Hex,
+      abi: [
+        {
+          inputs: [
+            {
+              components: [
+                { internalType: "address", name: "sender", type: "address" },
```

Comment:
Can reduce loc here with parseAbi from viem, but need to confirm as the params are not straight forward.
Just check if it can be otherwise looks good

```ts
const parsedAbi = parseAbi(["function getUserOpHash(...)"]);
const epContract = getContract({
   address: entryPointAddress as Hex,
   abi: parsedAbi,
   publicClient,
});
const count = await epContract.read.getUserOpHash([userOp]);
```